hawkular-inventory-api/src/main/java/org/hawkular/inventory/base/Associator.java

```diff
@@ -65,7 +65,7 @@ public Relationship associationWith(Path path) throws RelationNotFoundException
     }
 
     protected void checkPathLegal(Path targetPath) {
-        if (!context.entityClass.equals(targetPath.getSegment().getElementType())) {
+        if (!context.entityClass.getSimpleName().equals(targetPath.getSegment().getElementType().getSimpleName())) {
```

Comment:
Yes, the string comparison is not ideal both here and also in `SegmentType.fromElementType(Class<?> elementType)` https://github.com/ppalaga/hawkular-inventory/commit/7256fb780019a62dc14d542c3890ad118c24c5f8#diff-2ced6bb7b8b179c9c87ce45e3c09c837R119

Thinking of it as consequently as I can, I'd say that the most efficient fix would be to use `SegmentType` everywhere instead of `Class`. I started doing it at some point but then I have given up because it was quite a lot of work and I wanted to get a preliminary green light from you first. Thus, I have followed the strategy to make the this PR as less impact in the old inventory as possible. 

So how about using `SegmentType` everywhere instead of `Class`?